src/Commands/PartCommand.cpp

```diff
@@ -0,0 +1,66 @@
+#include "PartCommand.hpp"
+#include "Replies.hpp"
+#include <string>
+
+PartCommand::PartCommand(std::string source, std::vector<std::string> params,
+                         Client* client) {
+    checkParams(client, params);
+    parseParams(client, params);
+    this->_source = source;
+    this->_params = params;
+    this->_client = client;
+}
+
+PartCommand::~PartCommand() {}
+
+void PartCommand::checkParams(Client* client, std::vector<std::string> params) {
+    if (params.size() < 1) {
+        client->sendMessage(Replies::ERR_NEEDMOREPARAMS(client, "PART"));
+        throw;
+    }
+}
+
+void PartCommand::parseParams(Client* client, std::vector<std::string> params) {
+    std::istringstream iss(params[0]);
+    std::string chanName;
+    size_t i = 0;
+    this->_reason = "";
+
+    while (std::getline(iss, chanName, ',')) {
+        try {
+            _channels[i] = Server::getInstance().findChannel(chanName);
+        } catch (const Server::ChannelNotFoundException()) {
+            client->sendMessage(Replies::ERR_NOSUCHCHANNEL(client, chanName));
+            continue;
+        }
+        i++;
+    }
+    if (params.size() > 1) {
+        _reason = params[1];
+    }
+}
+
+std::string PartCommand::createReply(Channel* channel) {
+    std::string reply;
+
+    reply = ":" + _client->getNickname() + " PART " + channel->getName() + " " +
+            _reason;
+
+    return reply;
+}
+
+void PartCommand::run() {
+    for (size_t i = 0; i < _channels.size(); ++i) {
+        if (_channels[i]->isInChannel(*_client)) {
```

Comment:
I really don't like the fact tha isInChannel, kickOperator, eraseClient and all take a reference for a strange reason.
We need to maintain consistency, and the functions need to take a pointer since we always have pointers, it will remove the dereferencing, and every other functions in the code are doing this.

Reply:
will be addressed in a next pr